docs/src/fs.rst

```diff
@@ -227,6 +228,15 @@ API
         in binary mode. Because of this the O_BINARY and O_TEXT flags are not
         supported.
 
+.. c:function:: int uv_fs_openat(uv_loop_t* loop, uv_fs_t* req, uv_file file, const char* path, int flags, int mode, uv_fs_cb cb)
```

Comment:
Are there any cross platform differences? For example, are relative paths and absolute paths supported? I don't think we have test coverage for this, but from reading the docs:

`openat()`:
> If pathname is absolute, then dirfd is ignored.

`NtCreateFile()`'s `RootDirectory`:
> If this value is NULL, the ObjectName member must be a fully qualified file specification that includes the full path to the target file. If this value is non-NULL, the ObjectName member specifies a file name relative to this directory.

It seems like we need some extra logic on Windows to emulate the `openat()` behavior.

(Side note: last I saw, WASI didn't handle absolute paths, so it was easier for them to ignore inconsistencies like that. libuv needs to be a bit more thorough in that regard.)

Reply:
Good point! Added a test case for absolute path.